ROADMAP.md

```diff
@@ -0,0 +1,157 @@
+It is not intended to be all encompassing or a binding development plan, 
+but should be regarded as an indication of the focus of the core development team. 
+
+There are four main topic areas that will be focused on: 
+- Releasing OpenMDAO V3.0
+- Model & Data Visualization 
+- New Differentiation Tools that Reduce User Effort
+- Establishing A Community Contribution Process
+
+----------------------------------------------
+----------------------------------------------
+----------------------------------------------
+# Releasing OpenMDAO V3.0
+
+In January 2020, we plan to release V3.0 of the framework. 
+This update will **NOT** represent significant change to the codebase or functionality, 
+but will include several small but important practical changes: 
+
+## 1) Dropping support for Python 2.0 and supporting only Python 3.6 and greater
+- Scipy and Numpy have already dropped Python 2 support for future releases. 
+- Will allow us to modestly simplify our codebase 
+
+## 2) Removing all current deprecations 
+- We've made a strong attempt not to break backwards compatibility through deprecations where possible, 
+but V3.0 will solidify the current primary (i.e. non deprecated) APIs as the only ones. 
+
+
+----------------------------------------------
+----------------------------------------------
+----------------------------------------------
+# Model & Data Visualization 
+
+Motivation: Provide tools that make it easier to quickly build and debug complex 
+models with 100's of components organized into 10's of groups in 10's of hierarchy layers
+
+Overall Goal: Rely heavily on data stored in the CaseRecorder for all visualization. 
+This makes the visualization more portable 
+(i.e. a case database can be easily shared with others who could not necessarily run your model) 
+and also makes it available after a model was run. 
+
+## 1) Metamodel viewer   
+
+### Goal:
+    - Allow users to inspect their MetaModel to check its accuracy/smoothness/etc 
+    - Encourage greater use of OpenMDAO MetaModel components 
+
+### Potential Challenges:
+    - Current implementation is based on Bokeh, and performance isn't as fast as we'd like yet 
+    - Some functionality requires live prediction from MetaModel instance, 
+      so we require a server process running in background. 
+      This means the functionality can't be built into the CaseRecorder
+
+### Notes: 
+    - initial capability was released in OpenMDAO V2.9.0
+
+
+## 2) Improved n2 model viz tool  
+### Goal:
+    - Make the tool more intuitive for new users 
+    - Make the tool more useful for navigating models with deep hierarchies 
+      and large number of components/variables 
+
+### Potential Challenges:
+    - Current user interface is stretched to its limit, and can't 
+      integrate any expanded navigational features. We'll need a new concept 
+      for the UI 
```

Comment:
Proposed: JupyterLab extension. Some options exist:
- the ipywidgets road, a la [cadquery](https://github.com/bernhard-42/jupyter-cadquery) which would work with more clients
- a more "native" integration (not many public examples of really extreme customization... yet)

The end-user `nodejs` dependency can be avoided by distributing a CI-built (and tested) jupyterlab application on pypi/conda, either as part of the core, wholly-separate as `openmdaolab`, or half-way (for pip) with `openmdao[lab]`. We stopped just short of doing pypi releases of [robotlab](https://github.com/robots-from-jupyter/robotlab), but then went and built 700mb installers. Having full working installers with all the bells and whistles is not a terrible idea, but just-the-lab is likely more useful in this case.